scalyr_agent/copying_manager/worker.py

```diff
@@ -1021,6 +1036,16 @@ def __scan_for_new_bytes(self, current_time=None):
         for processor in self.__log_processors:
             processor.scan_for_new_bytes(current_time)
 
+    def get_full_checkpoints(self):  # type: () -> Dict[six.text_type, Any]
+        """
+        This method returns the full collection of the current log processor checkpoint states.
+        It also mainly designed to be called from the outside threads, so the checkpoint object should be protected
+        by the lock.
+        :return: the dict of the checkpoint states of the current log processors.
+        """
+        with self._checkpoints_lock:
+            return self._full_checkpoints
```

Comment:
Could also add a comment which says that this variable is updated when writing checkpoints in ``__write_checkpoint_state`` method.